src/lib/storage/pos_list.hpp

```diff
@@ -0,0 +1,135 @@
+#pragma once
+
+#include <utility>
+#include <vector>
+
+#include "types.hpp"
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+// For a long time, PosList was just a pmr_vector<RowID>. With this class, we want to add functionality to that vector,
+// more specifically, flags that give us some guarantees about its contents. If we know, e.g., that all entries point
+// into the same chunk, we can simplify things in split_pos_list_by_chunk_id.
+// Inheriting from std::vector is generally not encouraged, because the STL containers are not prepared for
+// inheritance. By making the inheritance private and this class final, we can assure that the problems that come with
+// a non-virtual destructor do not occur.
+
+struct PosList final : private pmr_vector<RowID> {
+ public:
+  using Vector = pmr_vector<RowID>;
+
+  using value_type = Vector::value_type;
+  using allocator_type = Vector::allocator_type;
+  using size_type = Vector::size_type;
+  using difference_type = Vector::difference_type;
+  using reference = Vector::reference;
+  using const_reference = Vector::const_reference;
+  using pointer = Vector::pointer;
+  using const_pointer = Vector::const_pointer;
+  using iterator = Vector::iterator;
+  using const_iterator = Vector::const_iterator;
+  using reverse_iterator = Vector::reverse_iterator;
+  using const_reverse_iterator = Vector::const_reverse_iterator;
+
+  /* (1 ) */ PosList() noexcept(noexcept(allocator_type())) {}
+  /* (1 ) */ explicit PosList(const allocator_type& allocator) noexcept : Vector(allocator) {}
+  /* (2 ) */ PosList(size_type count, const RowID& value, const allocator_type& alloc = allocator_type())
+      : Vector(count, value, alloc) {}
+  /* (3 ) */ explicit PosList(size_type count, const allocator_type& alloc = allocator_type()) : Vector(count, alloc) {}
+  /* (4 ) */ template <class InputIt>
+  PosList(InputIt first, InputIt last, const allocator_type& alloc = allocator_type())
+      : Vector(std::move(first), std::move(last)) {}
+  /* (5 ) */  // PosList(const Vector& other) : Vector(other); - Oh no, you don't.
+  /* (5 ) */  // PosList(const Vector& other, const allocator_type& alloc) : Vector(other, alloc);
+  /* (6 ) */ PosList(PosList&& other) noexcept
+      : Vector(std::move(other)), _references_single_chunk{other._references_single_chunk} {}
+  /* (6+) */ explicit PosList(Vector&& other) noexcept : Vector(std::move(other)) {}
+  /* (7 ) */ PosList(PosList&& other, const allocator_type& alloc)
+      : Vector(std::move(other), alloc), _references_single_chunk{other._references_single_chunk} {}
+  /* (7+) */ PosList(Vector&& other, const allocator_type& alloc) : Vector(std::move(other), alloc) {}
+  /* (8 ) */ PosList(std::initializer_list<RowID> init, const allocator_type& alloc = allocator_type())
+      : Vector(std::move(init), alloc) {}
+
+  PosList& operator=(PosList&& other) {
```

Comment:
why can we rely on the automatically generated copy-assignment operator, but not on the automatically generated move-assignment?

Reply:
We can.